Tests/PackageGraphPerformanceTests/DependencyResolverPerfTests.swift

```diff
@@ -0,0 +1,54 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright 2016 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import XCTest
+
+import Basic
+import PackageGraph
+
+import struct Utility.Version
+
+import TestSupport
+
+#if ENABLE_PERF_TESTS
+
+class DependencyResolverPerfTests: XCTestCase {
+
+    func testTrivalResolution_X500() throws {
```

Comment:
This test spends most time in resolver's `merge(constraints:intosubjectTo:excluding:)` and second most in VersionAssignmentSet's merging method